docs/Current/primary-cloud-for-web-applications.md

```diff
@@ -0,0 +1,147 @@
+4. Pricing can be higher for certain resources (such as container apps, realtime DB)
+5. Application insights provides fantastic functionality, but does not support GoLang natively as of today.
+
+**Risks** 
+1. Potentially higher costs for the same resource type
+2. Potentially less stable / performant for the same resource type (e.g. container apps)
+3. Need to migrate existing solutions
+
+**Benefits**  
+1. Easier to handle security, monitoring, cost control
+2. Reduce costs through more shared resources
+3. Reduce costs through requiring less "redundant" know-how
+
+### Alt 3: GCP-only strategy (for web apps)
+In this solution we use GCP as the primary platform for hosting web applications. In this scenario, Azure could still be used as a user directory and synchronized with GCP.
+
+GCP is developed by Google and in _general_ the following can be said of GCP:
+1. Low cost for realtime db (propriotory firebase)
+2. Good performance and low costs associated with hosting APIs (Cloud run)
+3. Good (better) security configuration "by default" (esp. things like cloudproxy)
+4. Pricing often slightly lower than Azure (with some notable exceptions like BigQuery, SpannerSQL).
+5. Monitoring and tracing tooling is less mature and more "clunky" than on Azure, but supports Go-lang out of the box.
+
+**Risks**  
+1. Need to migrate existing projects from Azure - many of which run on .Net (poorer experience)
+2. Smaller offering and missing features compared to Azure
+3. No presence in Norway (yet)
+4. Poorer monitoring experience
+5. Need to manage security, costs etc. in multiple clouds (due to corporate applications)
+
+**Benefits**  
+1. In general, cheaper and robust services (e.g. cloud run)
+2. Loved by existing teams
+
+### Alt 4: Low Cost hybrid strategy
+There are a number of alternative cloud providers such as Linode, Digital Ocean etc. which can provide competive pricing compared to Azure and GCP for certain types of resources. 
+
+This is an avenue we haven't fully investigated. In general, the service offerings are much smaller, so it is likely we'd need to use a combination of services. 
+
+**Risks**  
+1. More difficult to manage access control, onboarding/offboarding and maintining account access
+2. More things to document
+4. Typically less stable services (although there may be exceptions)
+5. Harder to monitor costs
+
+**Benefits**  
+1. Lower costs for certain resouce types
+
+## Assessment
+Intuitively an organization of our size and complexity should only use **one** cloud. However, since we already are spread accross two clouds and a range of technologies, the best solution becomes somewhat less clear cut.
+
+Specifically for end-user / high volume web applications, GCP is an appealing platform due to it's robust and secure nature. In addition, having applications that "talk" together in the same cloud is good for tracing, latency etc. 
```

Comment:
The "robustness" of CloudSQL is still an open point. The A-team project worked better after migrating to Azure than on GCP where the application / CloudSQL-db became unavailable for minutes at a time.

Reply:
Could this have been a configuration issue? I have not heard of this issue, which might mean that there wasn't any question to anyone that might know how to configure google cloud products.
It that regard I would say between the developers we currently have most actually know Google compared to azure.

Reply:
https://cloud.google.com/sql/docs/editions-intro
This is new and may be relevant. 99.95% and 99.99% SLA available.